src/prefect/docker/__init__.py

```diff
@@ -0,0 +1,20 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from prefect.docker.docker_image import DockerImage
+
+__all__ = ["DockerImage"]
+
+_public_api: dict[str, tuple[str, str]] = {
+    "DockerImage": ("prefect.docker.docker_image", "DockerImage"),
+}
+
+
+def __getattr__(name: str) -> object:
+    from importlib import import_module
+
+    if name in _public_api:
+        module, attr = _public_api[name]
+        return getattr(import_module(module), attr)
+
+    raise ImportError(f"module {__name__!r} has no attribute {name!r}")
```

Comment:
We used to have a `docker` module, but we moved it to `utilities.dockerutils` in https://github.com/PrefectHQ/prefect/pull/8788 because of https://github.com/PrefectHQ/prefect/issues/6519, so we might need to put `DockerImage` in `utilities.dockerutils`.

Reply:
Yeah, I think both those will help us. I'm cool with keeping `DockerImage` where it is in this PR. Seems like a low likelihood that we'll see the same problem.